lesson4/hometask_tohichU.rb

```diff
@@ -0,0 +1,23 @@
+class LazyHash < Hash
+
+  def method_missing(method)
+    if self.has_key?(method)
+      self[method].is_a?(Proc) ? self[method] = self[method].call : self[method]
+    else
+      super
+    end
+  end
+
+  def respond_to_missing?(method_name, include_private = false)
+    if self.has_key? method_name
+      true
+    else
+      super
+    end
+  end
+
+end
+
+
+lazy_hash = LazyHash.new
```

Comment:
Add more examples with commented result to check how we manage procs here. 
Like 
```ruby
p lazy_hash[:test]
p lazy_hash.test
p lazy_hash[:test]
```

Btw, @tohichU, where `#lazy?` method?